### PR TITLE
Adding tests for specific shape-related errors in BatchNorm and LayerNorm

### PR DESCRIPTION
**Pull request by BradLarson**
This adds two specific tests for BatchNorm and LayerNorm inference. These tests check for cases where changes to LayerNorm and BatchNorm had triggered shape mismatch errors in the MiniGo and Transformer models within [tensorflow/swift-models](https://github.com/tensorflow/swift-models).

The models now run cleanly, but these tests add checks against future regressions of this kind.

### CODE REVIEW

File: Tests/TensorFlowTests/LayerTests.swift [hunk 1]
```diff
@@ -741,6 +751,16 @@ final class LayerTests: XCTestCase {
             [11.077844 , 12.092919 ,  3.0350027,  4.7778125,  8.969137],
             accuracy: 1e-5)
     }
+    
+    func testLayerNormInference() {
+        Context.local.learningPhase = .inference
+        // This tests for a specific failure that had impacted the Transformer model.
+        let transformerTensor = Tensor<Float>(randomUniform: [1, 1, 768])
+        let transformerLayerNorm = LayerNorm(
+            featureCount: 768, axis: -1, epsilon: Tensor<Float>(1e-5))
```
Comment (eaplatanios) [inline Tests/TensorFlowTests/LayerTests.swift]: Minor suggestion just so it matches the style of the batch norm test.
```suggestion
            featureCount: 768,
            axis: -1,
            epsilon: Tensor<Float>(1e-5))
```

**Comment by eaplatanios** [comment]
Thanks @BradLarson! It's good to finally have those tests. :)
